classes/issue/maps/Schema.inc.php

```diff
@@ -0,0 +1,163 @@
+<?php
+
+namespace APP\issue\maps;
+
+use APP\core\Application;
+use APP\core\Services;
+use APP\facades\Repo;
+use APP\issue\Issue;
+use APP\issue\IssueGalleyDAO;
+use APP\journal\SectionDAO;
+use Illuminate\Support\Enumerable;
+use PKP\context\Context;
+use PKP\core\PKPRequest;
+use PKP\db\DAORegistry;
+use PKP\services\PKPSchemaService;
+use UserGroupDAO;
+
+class Schema extends \PKP\core\maps\Schema
+{
+    /** @var Enumerable */
+    public $collection;
+
+    /** @var string */
+    public $schema = PKPSchemaService::SCHEMA_ISSUE;
+
+    public function __construct(PKPRequest $request, Context $context, PKPSchemaService $schemaService)
+    {
+        parent::__construct($request, $context, $schemaService);
+    }
+
+    /**
+     * Map an Issue
+     *
+     * Includes all properties in the Issue schema
+     *
+     */
+    public function map(Issue $item): array
+    {
+        return $this->mapByProperties($this->getProps(), $item);
+    }
+
+    /**
+     * Summarize an Issue
+     *
+     * Includes properties with the apiSummary flag in the Issue schema.
+     *
+     */
+    public function summarize(Issue $item): array
+    {
+        return $this->mapByProperties($this->getSummaryProps(), $item);
+    }
+
+    /**
+     * Map a collection of Issues
+     *
+     * @see self::map
+     *
+     */
+    public function mapMany(Enumerable $collection): Enumerable
+    {
+        $this->collection = $collection;
+        return $collection->map(function ($item) {
+            return $this->map($item);
+        });
+    }
+
+    /**
+     * Summarize a collection of Issues
+     *
+     * @see self::summarize
+     *
+     */
+    public function summarizeMany(Enumerable $collection): Enumerable
+    {
+        $this->collection = $collection;
+        return $collection->map(function ($item) {
+            return $this->summarize($item);
+        });
+    }
+
+    /**
+     * Map schema properties of an Issue to an assoc array
+     *
+     */
+    private function mapByProperties(array $props, Issue $issue): array
+    {
+        $output = [];
+
+        foreach ($props as $prop) {
+            switch ($prop) {
+                case '_href':
+                    $output[$prop] = $this->getApiUrl('issues/' . $issue->getId(), $this->context->getData('urlPath'));
+                    break;
+                case 'articles':
+                    $data = [];
+
+                    $submissions = Repo::submission()->getMany(
+                        Repo::submission()
+                            ->getCollector()
+                            ->filterByContextIds([$issue->getJournalId()])
+                            ->filterByIssueIds([$issue->getId()])
+                    );
+
+                    /** @var UserGroupDAO $userGroupsDao */
+                    $userGroupsDao = DAORegistry::getDAO('UserGroupDAO');
+                    $userGroups = $userGroupsDao->getByContextId($this->context->getId());
```

Comment:
If this is run on a collection of 20 issues, it will retrieve the user groups 20 times. Extract this to a private method that stores the value and only runs it once:

```
private function getUserGroups()
{
    if (!isset($this->userGroups)) {
      $userGroupsDao = DAORegistry::getDAO('UserGroupDAO');
      $userGroups = $userGroupsDao->getByContextId($this->context->getId());
      $this->userGroups = $userGroups->toArray();
    }
    return $this->userGroups;
}
```